apps/build-onchain-apps/src/hooks/useBlockNumber.ts

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react';
+import { useNetwork } from 'wagmi';
+
+type BlockNumberResponse = {
+  block: number;
+};
+const useCurrentBlockNumber = (refreshIntervalMs = 10000) => {
+  const [blockNumber, setBlockNumber] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+  const { chain } = useNetwork();
+  const chainId = chain?.id;
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timeout;
+
+    const fetchBlockNumber = async () => {
+      if (!chainId) {
+        setBlockNumber(0);
+        return;
+      }
+
+      setIsLoading(true);
+
+      try {
+        const response = await fetch(`/api/chain/currentBlockNumber?chainId=${chainId}`);
+        if (!response.ok) {
+          throw new Error(`Error: ${response.status}`);
+        }
+
+        const data: BlockNumberResponse = (await response.json()) as BlockNumberResponse;
```

Comment:
```suggestion
        const data = (await response.json()) as BlockNumberResponse;
```
minor nit, but this should be inferred due to the cast

Reply:
fixed